[16.0][MIG] stock_release_channel: Migration to 16.0

lmignon opened the pull request.

lmignon committed:
[IMP] stock_release_channel: black, isort, prettier

lmignon committed:
[MIG] stock_release_channel: Migration to 16.0

rousseldenis commented:
/ocabot migration stock_release_channel

jbaudoux committed:
stock_release_channel: assign channel

When the move is not generated by a SO, also assign the release channel

jbaudoux committed:
stock_release_channel: multi warehouse

Organize release channels by warehouse.
Ease the configuration of release channel by picking types.

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+        for channel in self:
+            domain = self._field_picking_domains()["count_picking_released"]
+            domain += [("release_channel_id", "=", channel.id)]
+            released = self.env["stock.picking"].search(domain)
```
To avoid several search DB queries, I would have put this outside the loop and then filter on the release channel

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+            released = self.env["stock.picking"].search(domain)
+
+            if not released:
+                channel.picking_chain_ids = False
```
```suggestion
                channel.update({
                    "picking_chain_id": False,
                    "count_picking_chain": 0,
                    "count_picking_chain_in_progress": 0,
                    "count_picking_chain_done": 0,
                })
```

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+
+            self.env.cr.execute(*self._query_get_chain(released))
+            rows = self.env.cr.dictfetchall()
+            channel.picking_chain_ids = [row["picking_id"] for row in rows]
```
Same as above

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+            picking = self.env["stock.picking"].search(
+                domain, limit=1, order="date_done DESC"
+            )
+            channel.last_done_picking_id = picking
```
Same as above

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+            channel.last_done_picking_date_done = picking.date_done
+
+    def _default_sequence(self):
+        default_channel = self.env.ref(
```
Shouldn't it be better to set it as a configuration ?

rousseldenis commented on stock_release_channel/models/stock_release_channel.py:
```diff
@@ -0,0 +1,633 @@
+    def _action_move_for_field(self, field_domain, context=None):
+        domain = self._field_picking_domains()[field_domain]
+        domain += [("release_channel_id", "in", self.ids)]
+        pickings = self.env["stock.picking"].search(domain)
```
Why not passing the domain directly to action key? That will avoid one search

rousseldenis reviewed (commented):
Some comments

sbejaoui committed:
[IMP] - improve release channel form view by adding notebook

sbejaoui reviewed (approved):
Functional test

sbejaoui committed:
[IMP] - stock_release_channel: better inheritance of assign_release_channel

jbaudoux commented on stock_release_channel/__manifest__.py:
```diff
@@ -0,0 +1,35 @@
+    "website": "https://github.com/OCA/wms",
+    "depends": [
+        "web",
+        "sale_stock",
```
I think we don't need to depend on sales anymore since my change here:
[e334ddc0bece05fd562714205655c809e24a6caa](https://github.com/OCA/wms/commit/e334ddc0bece05fd562714205655c809e24a6caa)

lmignon replied:
The dependency is removed and everything is working fine.

lmignon committed:
[FIX] stock_release_channel: Remove dependency on sale

The dependency on sale addon is no more required thanks to changes in https://github.com/OCA/wms/commit/e334ddc0bece05fd562714205655c809e24a6caa

simahawk commented:
Same as #470

simahawk commented:
https://github.com/OCA/wms/pull/600

simahawk commented:
@lmignon can you rebase pls?

simahawk commented:
@lmignon after reading your comment on #470 I can rollback #600 if you want.

jbaudoux commented:
> @lmignon after reading your comment on #470 I can rollback #600 if you want.

yes please

lmignon commented:
@simahawk rebase done....

simahawk reviewed (approved):
LGTM

simahawk commented:
/ocabot merge nobump
